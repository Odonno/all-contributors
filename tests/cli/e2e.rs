use assert_fs::TempDir;
use color_eyre::eyre::{Error, Result};
use insta::{Settings, assert_snapshot};
use std::fs;

use crate::helpers::{
    InstaSettingsExtensions, add_contributions, copy_config_file, copy_simple_readme_file,
    create_cmd,
};

#[test]
fn add_new_contribution_to_existing_contributor_and_generate() -> Result<()> {
    let temp_dir = TempDir::new()?;

    copy_config_file(&temp_dir)?;
    copy_simple_readme_file(&temp_dir)?;

    add_contributions(temp_dir.path(), "tekacs", "ideas,bug")?;

    let mut cmd = create_cmd(&temp_dir)?;

    cmd.env("NO_COLOR", "1").arg("generate");

    cmd.assert().try_success()?;

    let updated_readme = fs::read_to_string(temp_dir.join("readme.md"))?;

    let mut insta_settings = Settings::new();
    insta_settings.add_cli_location_filter();
    insta_settings.bind(|| {
        assert_snapshot!(updated_readme);
        Ok::<(), Error>(())
    })?;

    temp_dir.close()?;

    Ok(())
}

#[test]
fn add_new_contributor_and_its_contributions_and_generate() -> Result<()> {
    let temp_dir = TempDir::new()?;

    copy_config_file(&temp_dir)?;
    copy_simple_readme_file(&temp_dir)?;

    add_contributions(temp_dir.path(), "odonno", "code,ideas")?;

    let mut cmd = create_cmd(&temp_dir)?;

    cmd.env("NO_COLOR", "1").arg("generate");

    cmd.assert().try_success()?;

    let updated_readme = fs::read_to_string(temp_dir.join("readme.md"))?;

    let mut insta_settings = Settings::new();
    insta_settings.add_cli_location_filter();
    insta_settings.bind(|| {
        assert_snapshot!(updated_readme);
        Ok::<(), Error>(())
    })?;

    temp_dir.close()?;

    Ok(())
}
