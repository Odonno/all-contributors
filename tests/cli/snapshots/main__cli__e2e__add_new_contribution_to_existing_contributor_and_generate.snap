---
source: tests/cli/e2e.rs
expression: updated_readme
---
<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore-start -->
<!-- markdownlint-disable -->
<table>
	<tbody>
		<tr>
			<td align="center" valign="top" width="14.29%"><a href="http://www.tekacs.com/"><img src="https://avatars.githubusercontent.com/u/63247?v=4?s=100" width="100px;" alt="Amar Sood"><br /><sub><b>Amar Sood</b></sub></a><br /><a href="https://github.com/Odonno/surrealdb-migrations/commits?author=tekacs" title="Code">💻</a> <a href="#ideas-tekacs" title="Ideas, Planning, & Feedback">🤔</a> <a href="https://github.com/Odonno/surrealdb-migrations/issues?q=author%3Atekacs" title="Bug reports">🐛</a></td>
			<td align="center" valign="top" width="14.29%"><a href="https://github.com/Roms1383"><img src="https://avatars.githubusercontent.com/u/21016014?v=4?s=100" width="100px;" alt="Rom's"><br /><sub><b>Rom's</b></sub></a><br /><a href="https://github.com/Odonno/surrealdb-migrations/commits?author=Roms1383" title="Code">💻</a> <a href="#ideas-Roms1383" title="Ideas, Planning, & Feedback">🤔</a> <a href="#infra-Roms1383" title="Infrastructure">🚇</a></td>
			<td align="center" valign="top" width="14.29%"><a href="https://github.com/pranaypratyush"><img src="https://avatars.githubusercontent.com/u/10587083?v=4?s=100" width="100px;" alt="Pranay Pratyush"><br /><sub><b>Pranay Pratyush</b></sub></a><br /><a href="https://github.com/Odonno/surrealdb-migrations/issues?q=author%3Apranaypratyush" title="Bug reports">🐛</a></td>
			<td align="center" valign="top" width="14.29%"><a href="https://github.com/ansarizafar"><img src="https://avatars.githubusercontent.com/u/5788284?v=4?s=100" width="100px;" alt="Zafar Ansari"><br /><sub><b>Zafar Ansari</b></sub></a><br /><a href="#ideas-ansarizafar" title="Ideas, Planning, & Feedback">🤔</a></td>
			<td align="center" valign="top" width="14.29%"><a href="https://turulix.de/"><img src="https://avatars.githubusercontent.com/u/28957605?v=4?s=100" width="100px;" alt="Tim"><br /><sub><b>Tim</b></sub></a><br /><a href="#ideas-turulix" title="Ideas, Planning, & Feedback">🤔</a></td>
			<td align="center" valign="top" width="14.29%"><a href="https://github.com/snarkipus"><img src="https://avatars.githubusercontent.com/u/3688989?v=4?s=100" width="100px;" alt="Matt Jackson"><br /><sub><b>Matt Jackson</b></sub></a><br /><a href="https://github.com/Odonno/surrealdb-migrations/issues?q=author%3Asnarkipus" title="Bug reports">🐛</a></td>
			<td align="center" valign="top" width="14.29%"><a href="https://github.com/keyruu"><img src="https://avatars.githubusercontent.com/u/53177682?v=4?s=100" width="100px;" alt="Lucas"><br /><sub><b>Lucas</b></sub></a><br /><a href="#ideas-keyruu" title="Ideas, Planning, & Feedback">🤔</a></td>
		</tr>
	</tbody>
</table>

<!-- markdownlint-restore -->
<!-- prettier-ignore-end -->

<!-- ALL-CONTRIBUTORS-LIST:END -->
